domain/secretbackend/bootstrap/bootstrap.go

```diff
@@ -22,31 +23,41 @@ import (
 // CreateDefaultBackends inserts the initial secret backends during bootstrap.
 func CreateDefaultBackends(modelType coremodel.ModelType) internaldatabase.BootstrapOpt {
 	return func(ctx context.Context, controller, model database.TxnRunner) error {
-		backendUUID, err := uuid.NewUUID()
-		if err != nil {
-			return errors.Trace(err)
-		}
-		return errors.Trace(controller.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-			upsertBackendStmt, err := sqlair.Prepare(`
-INSERT INTO secret_backend
-    (uuid, name, backend_type_id)
-VALUES ($SecretBackend.*)`, state.SecretBackend{})
+		err := controller.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+			err := createBackend(ctx, tx, juju.BackendName, domainsecretbackend.BackendTypeController)
 			if err != nil {
 				return errors.Trace(err)
 			}
-			backendName := juju.BackendName
-			backendType := domainsecretbackend.BackendTypeController
 			if modelType == coremodel.CAAS {
-				backendName = kubernetes.BackendName
-				backendType = domainsecretbackend.BackendTypeKubernetes
+				err := createBackend(ctx, tx, kubernetes.BackendName, domainsecretbackend.BackendTypeKubernetes)
+				if err != nil {
+					return errors.Trace(err)
+				}
 			}
-			err = tx.Query(ctx, upsertBackendStmt, state.SecretBackend{
-				ID:                  backendUUID.String(),
-				Name:                backendName,
-				BackendTypeID:       backendType,
-				TokenRotateInterval: internaldatabase.NullDuration{},
-			}).Run()
-			return errors.Annotatef(err, "cannot create secret backend %q", backendName)
-		}))
+			return nil
+		})
+		return domain.CoerceError(err)
```

Comment:
With the recent PR to automatically coerce txn errors, check if we still need this here